src/hostname/hostnamed.c

```diff
@@ -868,6 +904,33 @@ static int context_write_data_machine_info(Context *c) {
         return 0;
 }
 
+static int property_get_ansi_color(
+                sd_bus *bus,
+                const char *path,
+                const char *interface,
+                const char *property,
+                sd_bus_message *reply,
+                void *userdata,
+                sd_bus_error *error) {
+
+        Context *c = ASSERT_PTR(userdata);
+        _cleanup_free_ char *color = NULL;
+        int r;
+
+        context_read_etc_hostname(c);
+        context_read_machine_info(c);
+
+        if (isempty(c->data[PROP_ANSI_COLOR]))
+                color = fallback_ansi_color(c);
+        else {
+                r = asprintf(&color, "\x1B[%sm", c->data[PROP_ANSI_COLOR]);
```

Comment:
the "\x1B[` really should be prefixed client-side, and the "m" should be suffixed client-side too